### ISSUE TITLE
Is the Current Validation Loss Correct?

### ISSUE THREAD

**Issue by lapp0** [KellerJordan/modded-nanogpt#32]
Current record has validation loss based on a sliding window / `attn_blocksize` than previous records.

```
attn_blocksize = torch.tensor(64*((step/args.num_iterations * (1792 - 64) + 64)//64), dtype=torch.int, device='cuda')
...
val_loss += model(x_val, y_val, attn_blocksize=attn_blocksize)
```

When running the trainer, once complete the `attn_blocksize` will be `1728`:

```
>>> 64*((1750/1792 * (1792 - 64) + 64)//64)
1728.0
```

A sliding window size of 1728 is inconsistent with previous runs which have used [1024](https://github.com/KellerJordan/modded-nanogpt/blob/master/records/111924_FlexAttention/8384493d-dba9-4991-b16b-8696953f5e6d.txt#L259)

Early token entropy is higher than later token entropy, meaning the validation loss may be biased.

Am I correct in that this was miscalculated by the submitter and instead we should validate with `model(x_val, y_val, attn_blocksize=1024)`?

**Comment by KellerJordan** [comment]
you're right that previous runs used 1024 and this run maxes out at 1728. any window size is OK - you could go even higher if that performs better (but it doesn't seem to)
